server/opendp_apps/terms_of_access/models.py

```diff
@@ -21,6 +33,30 @@ class Meta:
     def __str__(self):
         return f'{self.name} - v{self.version}'
 
+    def save(self, *args, **kwargs):
+        """
+        Save a TermsOfAccess model.
+        :param args:
+        :param kwargs: Potentially contains template_path
+        :return:
+        """
+        # For testing and flexibility, we should allow this method to accept an argument `template_path` which
+        # specifies the location of the template to be compared against.
+        try:
+            template_path = kwargs.pop('template_path')
+        except KeyError:
+            template_path = os.path.dirname(os.path.realpath(__file__)) + f'/templates/{self.version}.html'
```

Comment:
- Add more protection on opening a file, e.g. it shouldn't be from an arbitrary starting directory or allow `../` etc
- Ties into previous comment about passing the template_path
   - Can there be file uploads? Or a "template_name" field addd to the model with choices restricted to all ".html" files from a single directory?  (?)

Reply:
The "arbitrary starting directory" can only be used within tests, it can't be accessed via the admin console. We shouldn't allow file uploads, because the point is to make sure the templates are in version control under a single directory, and are not being changed once they are saved in the database.